packages/compass/src/main/auto-update-manager.ts

```diff
@@ -15,21 +236,114 @@ const API_PRODUCT: Record<string, string> = {
   'mongodb-compass-readonly': 'compass-readonly',
 };
 
+export type AutoUpdateManagerOptions = {
+  endpoint: string;
+  platform: string;
+  arch: string;
+  product: string;
+  channel: string;
+  version: string;
+  updateCheckInterval: number;
+  initialUpdateDelay: number;
+};
+
+const emitter = new EventEmitter();
+
 class CompassAutoUpdateManager {
+  private constructor() {
+    // marking constructor as private to disallow usage
+  }
+
   private static initCalled = false;
+  private static state = AutoUpdateManagerState.Initial;
 
-  private static _init(): void {
-    log.info(
-      mongoLogId(1001000130),
-      'CompassAutoUpdateManager',
-      'Initializing'
+  static autoUpdateOptions: AutoUpdateManagerOptions;
+
+  static getFeedURLOptions(): FeedURLOptions {
+    const { endpoint, product, channel, platform, arch, version } =
+      this.autoUpdateOptions;
+
+    return {
+      url: `${endpoint}/api/v2/update/${product}/${channel}/${platform}-${arch}/${version}`,
+    };
+  }
+
+  static getUpdateCheckURL() {
+    const { endpoint, product, channel, platform, arch, version } =
+      this.autoUpdateOptions;
+
+    return `${endpoint}/api/v2/update/${product}/${channel}/${platform}-${arch}/${version}/check`;
+  }
+
+  static async checkForUpdate(): Promise<{
+    name: string;
+    from: string;
+    to: string;
+  } | null> {
+    try {
+      const result = await got(this.getUpdateCheckURL());
+      if (result.statusCode !== 200) {
```

Comment:
should we log stuff like `> 400` as errors?

Reply:
Ah, good catch, didn't thought of that, let me check

Reply:
Yeah, it will throw for these codes: https://github.com/sindresorhus/got/blob/5f278d74125608b7abe75941cb6a71e21e0fb892/source/core/response.ts#L116-L121